main.go

```diff
@@ -95,18 +98,32 @@ func main() {
 
 	image := strings.SplitN(newTag, ":", 2)[0]
 
-	err := filepath.Walk(templatePath, func(path string, f os.FileInfo, err error) error {
+	err := filepath.Walk(templatePath, func(path string, info os.FileInfo, err error) error {
 		if strings.HasSuffix(path, ".yaml") {
-			data, err := ioutil.ReadFile(path)
+			f, err := os.Open(path)
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "Failed to read file %s - skipping.\n", path)
 			} else {
-				t := miniDeployment{}
-				err = yaml.Unmarshal(data, &t)
-				if err != nil {
-					fmt.Fprintf(os.Stderr, "Failed to parse YAML in %s - skipping.\n", path)
-				} else {
-					investigateFile(t, path, image, newTag)
+				defer f.Close()
```

Comment:
After reading https://tour.golang.org/flowcontrol/12 I think that maybe it's not ok to call Defer here.  Perhaps I should move the block from line 102 should into a function?

Reply:
Well, it's correct to place it here, but the effect will be quite limited as it will be called when main() exits. But I wouldn't bother changing the structure of the code to avoid a leak, that won't be a leak 1 ms later when the app finishes.